message.go

```diff
@@ -25,13 +35,13 @@ func decodeToNull(dec decoder, m *message) error {
 	var err error
 	for err == nil && m.remainingFields > 0 {
 		i := new(interface{})
-		decodeMessage(dec, m, i)
+		err = decodeMessage(dec, m, i)
 	}
 	return err
 }
 
 func decodeError(dec decoder, m *message, f ErrorUnwrapper) (appErr error, dispatchErr error) {
-	var s string
+	var s string = ""
```

Comment:
nit: this isn't really idiomatic -- i'd expect either `var s string` (since relying on zero-initialization seems more idiomatic) or `s := ""`, but not both.

Also, may as well move this down to be closer to where it's used